tape/src/test/java/com/squareup/tape/QueueFileTest.java

```diff
@@ -68,6 +68,41 @@
     assertThat(queue.peek()).isEqualTo(expected);
   }
 
+  @Test public void testAddElementTooLarge() throws IOException {
+    QueueFile queue = new QueueFile(file);
+    byte[] newBigValue = new byte[QueueFile.MAX_ELEMENT_SIZE + 1];
+    try {
+      queue.add(newBigValue);
+      fail("Should have failed with IllegalArgumentException for element too large.");
+    } catch (IllegalArgumentException e) {
+      assertThat(e).hasMessage("Adding element that is " +
+          (QueueFile.MAX_ELEMENT_SIZE + 1) + " to queue with max"
+          + " element size " + QueueFile.MAX_ELEMENT_SIZE + ".");
+    } finally {
+      queue.close();
+    }
+  }
+
+  @Test public void testAddElementToQueueAboutToOverrun() throws IOException {
+    QueueFile queue = new QueueFile(file);
+    final int elementSize = QueueFile.MAX_ELEMENT_SIZE / 2;
+    byte[] expected = new byte[elementSize];
+    for (int i = 0; i < (Integer.MAX_VALUE / 2 / elementSize - 1); i++) {
+      queue.add(expected);
+    }
+    try {
+      queue.add(expected);
+      fail("Should have failed with IllegalArgumentException for element too large.");
+    } catch (IllegalArgumentException e) {
+      assertThat(e).hasMessage("Cannot add data of length " + elementSize + " to the "
+          + "queue which already has a length of " + (Integer.MAX_VALUE / 2 + 1)
+          + ". The max file size is "
+          + "1,073,741,823 (half of Integer.MAX_VALUE).");
```

Comment:
nit: generally avoid asserting on the full details of an exception message as it makes things hard to change later. Focus on the the pieces that are interesting in the message.

Reply:
I dont think this FEST assertion library has an Exception 'messageContains' method, so i'll have to pull out the message first.

Reply:
Just leave the full message.